derp/derphttp/derphttp_client.go

```diff
@@ -435,6 +459,19 @@ func (c *Client) connect(ctx context.Context, caller string) (client *derp.Clien
 		if resp.StatusCode != http.StatusSwitchingProtocols {
 			b, _ := io.ReadAll(resp.Body)
 			resp.Body.Close()
+
+			// Only on StatusCode < 500 in case a gateway timeout
+			// or network intermittency occurs!
+			if resp.StatusCode < 500 {
```

Comment:
Is this the only status code it could be? I could see a proxy just dropping the connection with no status if it didn't like the Upgrade, or returning `400` perhaps?

Reply:
`400` seems to be the _standard_ status code if an upgrade fails, but I'm not sure we can guarantee that with all load balancers that clouds may have. That's why I chose < 500 because it'd generally mean that the server intentionally chose to not accept the WebSocket for some reason, which `coderd` will never do.